### PR TITLE
[SFN] [IntrinsicFunctions]: snapshot tests, Array functions, String functions, fixes

### PR DESCRIPTION
**Pull request by MEPalma**
In this PR:
- Snapshot tests for SFN Intrinsic Functions
- Fixes for current Intrinsic Functions
- Base implementation of States Array functions
- Base implementation of States String operations functions

### CODE REVIEW

File: localstack/services/stepfunctions/asl/component/intrinsic/function/statesfunction/generic/string_format.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+        string_format_parts: list[str] = string_format.split(self._DELIMITER)
+        string_result: str = ""
+        for part in string_format_parts:
+            string_result += part
+            if values:
+                value = values.pop()
+                if not isinstance(value, str):
+                    value = self._json_to_literal(value)
+                string_result += value
```
Comment (dominikschubert) [inline localstack/services/stepfunctions/asl/component/intrinsic/function/statesfunction/generic/string_format.py:57]: Could be refactored a bit to be more idiomatic IMO.

A suggestions/nits:
- check the length of values & placeholders explicitly.
- first fix all the values (_json_to_literal) in the same step as you're reversing it, and then either doing the string-formatting loop it in a simpler imperative style with repeated `replace` calls (e.g. `string_format = string_format.replace("{}", value, 1)` or in a maybe more pythonic way via list expressions where you get a list of parts without the placeholders (your `string_format_parts`) and the values, zip  them and then just join them into a string again.

**Review by dominikschubert — APPROVED** [review]
Looks good already :rocket: 

I've added a few comments with suggestions for improvements or where I had questions/other related remarks.

Love the attention to detail for the test cases :+1:

File: localstack/services/stepfunctions/asl/component/intrinsic/function/statesfunction/array/array_partition.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+            raise ValueError(
+                f"Expected 2 arguments for function type '{type(self)}', but got: '{arg_list}'."
+            )
+
+    def _eval_body(self, env: Environment) -> None:
+        self.arg_list.eval(env=env)
+        chunk_size = env.stack.pop()
+        if not isinstance(chunk_size, (int, float)):
+            raise ValueError(f"Expected an integer value as chunk_size, but got {chunk_size}.")
+        chunk_size = round(chunk_size)
```
[the remark this answers is not in the document]
Comment (MEPalma) [reply localstack/services/stepfunctions/asl/component/intrinsic/function/statesfunction/array/array_partition.py:46]: we've taken this from the documentation, here https://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-intrinsic-functions.html under States.PartitionArray, arguments paragraph it says 'If you specify a non-integer value for the second argument, Step Functions will round it off to the nearest integer.'.

File: localstack/services/stepfunctions/asl/component/intrinsic/function/statesfunction/array/array_unique.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    def _eval_body(self, env: Environment) -> None:
+        self.arg_list.eval(env=env)
+
+        array = env.stack.pop()
+        if not isinstance(array, list):
+            raise ValueError(f"Expected an array type, but got '{array}'.")
+
+        items_set = set(array)
+        unique_array = list(items_set)
+        env.stack.append(unique_array)
```
[the remark this answers is not in the document]
Comment (MEPalma) [reply localstack/services/stepfunctions/asl/component/intrinsic/function/statesfunction/array/array_unique.py:54]: Great point, I missed this! I updated this logic to ensure it is always sorted (according to the input list).

**Pull request merged**